File: runtime/frontend/assets/js/dashboard/jobs.js
(() => {
  const list = document.getElementById('jobs-list');
  const renderJobs = (jobs) => {
    if (!jobs.length) {
      list.innerHTML = '<div class="empty-card">No jobs available.</div>';
      return;
    }
    list.innerHTML = jobs.map((job) => `
      <a class="summary-card" href="/dashboard/jobs/${job.job_id}">
        <div class="summary-row"><span>Job</span><span>${job.job_id}</span></div>
        <div class="summary-row"><span>Status</span><span class="badge">${job.status}</span></div>
      </a>
    `).join('');
  };
  fetch('/api/dashboard/jobs', { headers: { Accept: 'application/json' } })
    .then((res) => res.json().then((json) => [res.ok, json]))
    .then(([ok, data]) => {
      if (!ok) {
        list.innerHTML = `<div class="error-card">Failed to load jobs. request_id: ${data.request_id || 'n/a'}</div>`;
        return;
      }
      renderJobs(data.jobs || []);
    })
    .catch((err) => {
      list.innerHTML = `<div class="error-card">${String(err)}</div>`;
    });
})();

File: runtime/crates/api/src/web/dashboard/jobs.rs
use bominal_ui::{DashboardSection, render_dashboard_bottom_nav};

use super::super::{app_shell_topbar, dashboard_desktop_sidebar, html_escape};
pub fn render_dashboard_jobs(email: &str) -> String {
    let topbar = app_shell_topbar("Jobs", &format!("Signed in as {}", html_escape(email)));
    let sidebar = dashboard_desktop_sidebar("jobs");
    let bottom_nav = render_dashboard_bottom_nav(DashboardSection::Jobs);
    format!(
        r#"{topbar}
<main class="mx-auto w-full max-w-[480px] px-4 pb-24 pt-4 md:max-w-7xl md:px-6 md:pt-8">
  <div class="md:grid md:grid-cols-[220px_minmax(0,1fr)] md:items-start md:gap-4">
    {sidebar}
    <section class="glass-card rounded-[22px] p-5">
      <h2 class="text-lg font-semibold txt-strong">My runtime jobs</h2>
      <div id="jobs-list" class="mt-4 space-y-2"><div class="loading-card">Loading jobs...</div></div>
    </section>
  </div>
</main>
{bottom_nav}
<script>
(() => {{
  const list = document.getElementById('jobs-list');
  const renderJobs = (jobs) => {{
    if (!jobs.length) {{
      list.innerHTML = '<div class="empty-card">No jobs available.</div>';
      return;
    }}
    list.innerHTML = jobs.map((job) => `
      <a class="summary-card" href="/dashboard/jobs/${{job.job_id}}">
        <div class="summary-row"><span>Job</span><span>${{job.job_id}}</span></div>
        <div class="summary-row"><span>Status</span><span class="badge">${{job.status}}</span></div>
      </a>
    `).join('');
  }};
  fetch('/api/dashboard/jobs', {{ headers: {{ Accept: 'application/json' }} }})
    .then((res) => res.json().then((json) => [res.ok, json]))
    .then(([ok, data]) => {{
      if (!ok) {{
        list.innerHTML = `<div class="error-card">Failed to load jobs. request_id: ${{data.request_id || 'n/a'}}</div>`;
        return;
      }}
      renderJobs(data.jobs || []);
    }})
    .catch((err) => {{
      list.innerHTML = `<div class="error-card">${{String(err)}}</div>`;
    }});
}})();
</script>"#
    )
}
